chemprop/cli/fingerprint.py

```diff
@@ -16,10 +23,191 @@ class FingerprintSubcommand(Subcommand):
+        )
+
+        return parser
 
     @classmethod
     def func(cls, args: Namespace):
-        pass
+        args = process_common_args(args)
+        validate_common_args(args)
+        args = process_fingerprint_args(args)
+        main(args)
+
+
+def process_fingerprint_args(args: Namespace) -> Namespace:
+    if args.test_path.suffix not in [".csv"]:
+        raise ArgumentError(
+            argument=None, message=f"Input data must be a CSV file. Got {args.test_path}"
+        )
+    if args.output is None:
+        args.output = args.test_path.parent / (args.test_path.stem + "_fps.csv")
+    if args.output.suffix not in [".csv", ".pkl", ".pckl", ".pickle", ".npz"]:
+        raise ArgumentError(
+            argument=None, message=f"Output must be a CSV, Pickle, or npz file. Got {args.output}."
+        )
+    return args
+
+
+def find_models(model_path: Path):
+    if model_path.suffix in [".ckpt", ".pt"]:
+        return [model_path]
+    elif model_path.is_dir():
+        return list(model_path.rglob("*.ckpt")) + list(model_path.rglob("*.pt"))
+
+
+def make_fingerprint_for_model(
+    args: Namespace, model_path: Path, multicomponent: bool, output_path: Path
+):
+    model, input_scalers, output_scaler = load_model(
+        model_path, multicomponent
+    )  # TODO: connect input_scalers and output_scaler to the model
+    model.eval()
+
+    bounded = any(
+        isinstance(model.criterion, LossFunctionRegistry[loss_function])
+        for loss_function in LossFunctionRegistry.keys()
+        if "bounded" in loss_function
+    )
+
+    format_kwargs = dict(
+        no_header_row=args.no_header_row,
+        smiles_cols=args.smiles_columns,
+        rxn_cols=args.reaction_columns,
+        target_cols=None,
+        ignore_cols=None,
+        weight_col=None,
+        bounded=bounded,
+    )
+    featurization_kwargs = dict(
+        features_generators=args.features_generators, keep_h=args.keep_h, add_h=args.add_h
+    )
+
+    test_data = build_data_from_files(
+        args.test_path,
+        **format_kwargs,
+        p_descriptors=args.descriptors_path,
+        p_atom_feats=args.atom_features_path,
+        p_bond_feats=args.bond_features_path,
+        p_atom_descs=args.atom_descriptors_path,
+        **featurization_kwargs,
+    )
+    logger.info(f"test size: {len(test_data[0])}")
+    test_dsets = [
+        make_dataset(d, args.rxn_mode, args.multi_hot_atom_featurizer_mode) for d in test_data
+    ]
+
+    if multicomponent:
+        test_dset = data.MulticomponentDataset(test_dsets)
+    else:
+        test_dset = test_dsets[0]
+
+    X_d_scaler = input_scalers.get("X_d", None) if input_scalers else None
+    V_f_scaler = input_scalers.get("V_f", None) if input_scalers else None
+    E_f_scaler = input_scalers.get("E_f", None) if input_scalers else None
+    V_d_scaler = input_scalers.get("V_d", None) if input_scalers else None
```

Comment:
you check `if input_scalers` 4 times in a row here. Why not just check once and put everything in inside an `if`-block?

Reply:
Thank you for pointing that out! I followed the code in predict.py to write this module. I just modified the code in this file.